modules/core/shared/src/test/scala/gem/arb/ProperMotion.scala

```diff
@@ -30,5 +32,10 @@ trait ArbProperMotion {
       } yield ProperMotion(cs, ap, pv, rv, px)
     }
 
+  implicit val cogProperMotion: Cogen[ProperMotion] =
+    Cogen[(Coordinates, Epoch, Option[Offset], Option[RadialVelocity], Option[Angle])].contramap { p =>
+      (p.baseCoordinates, p.epoch, p.properVelocity, p.radialVelocity, p.parallax)
+    }
+
```

Comment:
Is it a scalacheck requirement (or recommendation to find the simplest failing test case) to define a `Cogen` for `ProperMotion` in order to run the Arbitrary tests? If that's not the case where will the Cogen instance be used?

Reply:
`Shrink` is used to minimize test cases but you don't normally need to specify it.

`Cogen` is used to generate interesting arbitrary functions. Given `Cogen[A]` and `Gen[B]` you can get `Gen[A => B]`. The idea is that `Cogen` takes an `A` and returns a hash that can be absorbed into the random seed that's passed on to the `Gen[B]`, so the incoming `A` helps to determine the returned `B`. Basically `type Cogen[A] = A => Long`. This [talk](https://www.youtube.com/watch?v=UlZGZh7hfbs) by Erik Osheim explains it very well starting around 36:00.